test/unit/repositories/test_catalogue_category.py

```diff
@@ -27,1334 +37,1204 @@
-    database_mock.catalogue_categories.update_one.assert_not_called()
+        assert self._created_property == self._expected_property_out
+
+    def check_create_property_failed_with_exception(self, message: str):
+        """Checks that a prior call to `call_create_property_expecting_error` worked as expected, raising an exception
+        with the correct message"""
+
+        self.catalogue_categories_collection.update_one.assert_not_called()
+
+        assert str(self._create_exception.value) == message
 
 
-def test_update_property_with_invalid_property_id(database_mock, catalogue_category_repository):
-    """
-    Test update_property performs the correct database update query when given an invalid property ID
-    """
+class TestCreateProperty(CreatePropertyDSL):
+    """Tests for creating a property"""
+
+    def test_create_property(self):
+        """Test creating a property in an existing catalogue category"""
+
+        self.mock_create_property(CATALOGUE_CATEGORY_PROPERTY_NUMBER_NON_MANDATORY_WITH_UNIT)
+        self.call_create_property(str(ObjectId()))
+        self.check_create_property_success()
+
+    def test_create_property_with_invalid_id(self):
+        """Test creating a property in a catalogue category with an invalid id"""
+
+        self.mock_create_property(CATALOGUE_CATEGORY_PROPERTY_NUMBER_NON_MANDATORY_WITH_UNIT)
+        self.call_create_property_expecting_error("invalid-id", InvalidObjectIdError)
+        self.check_create_property_failed_with_exception("Invalid ObjectId value 'invalid-id'")
+
+
+class UpdatePropertyDSL(CreatePropertyDSL):
+    """Base class for update property tests"""
+
+    _updated_property: CatalogueCategoryOut
+    _property_id: str
+    _update_exception: pytest.ExceptionInfo
+
+    def mock_update_property(self, property_in_data: dict):
+        """Mocks database methods appropriately to test the `update_property` repo method
+
+        :param property_in_data: Dictionary containing the catalogue category property data as would be required for a
+                                 CatalogueCategoryPropertyIn database model
+        """
+
+        self._property_in = CatalogueCategoryPropertyIn(**property_in_data)
+        self._expected_property_out = CatalogueCategoryPropertyOut(**self._property_in.model_dump(by_alias=True))
+
+        RepositoryTestHelpers.mock_update_one(self.catalogue_categories_collection)
+
+    def call_update_property(self, catalogue_category_id: str, property_id: str):
+        """Calls the CatalogueCategoryRepo `update_property` method with the appropriate data from a prior call to
+        `mock_update_property`
+        """
+
+        self._catalogue_category_id = catalogue_category_id
+        self._property_id = property_id
+        self._updated_property = self.catalogue_category_repository.update_property(
+            catalogue_category_id, property_id, self._property_in, session=self.mock_session
+        )
```

Comment:
I get the following warning here `Expected type 'CatalogueCategoryOut', got 'CatalogueCategoryPropertyOut' instead `

Reply:
Will fix on https://github.com/ral-facilities/inventory-management-system-api/pull/315.